instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -99,8 +102,48 @@ public static boolean isRegistered() {
     return REGISTERED.get();
   }
 
+  /**
+   * Register underlying driver that wasn't registered against {@link DriverManager}. This is mainly
+   * useful when drivers are initialized at build time, and you don't want to postpone class
+   * initialization (f.e. runtime initialization could make driver incompatible with other systems).
+   * Drivers registered with this method has higher priority over those registered against {@link
+   * DriverManager}.
+   *
+   * @param driver {@link Driver} that should be registered
+   */
+  public static void registerDriver(Driver driver) {
+    if (driver != null) {
+      DRIVERS.add(driver);
+    }
+  }
+
+  /**
+   * Unregister driver added via {@link #registerDriver(Driver)}.
+   *
+   * @param driver {@link Driver} that should be unregistered
+   * @return true if the driver was unregistered
+   */
+  public static boolean deregisterDriver(Driver driver) {
```

Comment:
removeDriverCandidate() ?
Some questions...
What will happen if that driver has already been used in the connection? 
Will we also need to de-register it from the DriverManager?

Reply:
> removeDriverCandidate() ?

Sure thanks.

> Some questions...
> What will happen if that driver has already been used in the connection?

The mechanism is same as with DriverManager, you select fist matching mechanism that accepts the URL. Drivers can change over time (not that I think it's common practice, but code allows it), so we can't take into account what driver was used previously.

> Will we also need to de-register it from the DriverManager?

We never register the driver with DriverManager, basically new "registry" is primary source of drivers, if driver candidate is not there, we use DriverManager. This should work around need to register drivers only used by OpenTelemetryDriver globally (where it may/or may not be desired):
- according to JDBC specs they should register themselves at class loading and registering drivers isn't our role (it's also not something you expect from OT driver)

Reply:
Thanks Michal.

Reply:
Also renamed.